### PR TITLE
Issue 2908 2

### LINKED ISSUE #1563: Parameter Revision - Modulateables
We intend to re-classify many parameters as modulateables. @nonlinear-labs when the concept is established and all re-classifications are known, the ParameterList document can be placed within this issue.

### LINKED ISSUE #2908: MC-Modulationsmöglichkeit für fast alle Parameter
In der folgenden Parameter Tabelle sind alle neu hinzugekommenen Modulationsziele grün markiert.

[ParameterList_1.7B neue Targets ssc.ods](https://github.com/nonlinear-labs-dev/C15/files/7507854/ParameterList_1.7B.neue.Targets.ssc.ods)

Nicht moduliert werden sollen:
- Velocity Amounts
- Key Tracking und Key Panning Amounts
- Unison Voices
- Mono Modes

Audio-Störungen können auftreten bei schlecht oder nicht gesmoothten Parametern wie Reverb - Pre Delay.
Bei Verbesserungen im Smoothing müssen wir die Auswirkungen auf die CPU-Last beobachten.

### LINKED ISSUE #2919: [Local: Off] Provide Info Texts for secondary HW Source Parameters
### Attention
We have to be careful about the order of changing the ParameterList.ods, so #2874 and #2916 need to be merged first. *(the file is binary and parallel, branch-based changes cannot be merged by gh)*

### Goal
When in MIDI Mode Local OFF, the secondary HW Parameters should be visible. As they are displayed as ordinary Parameters, we require Info Texts for them.

### Proposition
~~~
When the MIDI Local Setting is disabled, this control is the
parameter representation of the "Pedal 1" Hardware Source.
It reflects changes that will be sent when the MIDI Channels
Settings allow for MIDI to be sent without affecting the Synth.

This parameter will be visible next to the internal representation
of "Pedal 1" (which reflects changes from the User Interface or
MIDI Input, only affecting the Synth).
~~~

@nonlinear-labs, @davidm2894 please discuss further in the comments.

### LINKED ISSUE #2920: Erweiterte Bereiche für einige Key Trackings, Key Panning, einige Velocities, Unison Detune
- [x] Output Mixer - Key Pan: -100.0 ... 100.0 % (auch in #2756)
- [x] Oscillator A - Pitch - Key Trk: 0.00 ... 200.00 %
- [x] Oscillator B - Pitch - Key Trk: 0.00 ... 200.00 %
- [x] Comb Filter - Pitch - Key Trk: 0.00 ... 200.00 %
- [x] Comb Filter - AP Tune - Key Trk: 0.0 ... 200.0 %
- [x] Comb Filter - Hi Cut - Key Trk: 0.0 ... 200.0 %
- [x] SV Filter - Cutoff - Key Trk: 0.0 ... 200.0 %
- [x] Unison - Detune: 0.00 ... 24.00 st
- [x] Envelope A - Attack Velocity: -100.0 ... 100.0 %
- [x] Envelope B - Attack Velocity: -100.0 ... 100.0 %
- [x] Envelope C - Attack Velocity: -100.0 ... 100.0 %
- [x] Envelope A - Time Key Trk: -100.0 ... 100.0 % ~~(intuitiver, wenn man die bisherigen 0...100% auf 0...-100% mappt?)~~ 
- [x] Envelope B - Time Key Trk: -100.0 ... 100.0 %
- [x] Envelope C - Time Key Trk: -100.0 ... 100.0 %

Envelope X Attack - Velocity -> DB_t instead of %

### LINKED ISSUE #2921: Shaper A/B - FB - Env C: Bereich auf -100 % ...100 % erweitern. Negativer Teil dient zum Bypassen des Gates
100 % : Envelope C voll wirksam
0% : Gate voll wirksam
-100 %: Gate unwirksam, Signalweg permanent offen (Multiplikation mit 1)

Im positiven Bereich crossfaden wir - wie bisher - zwischen dem Gate-Signal und dem Envelope-Signal. Im negativen Bereich zwischen dem Gate-Signal und der konstanten 1.

Das Gate hilft, ein Dauer-Feedback (Durchpfeifen) beim Loslassen der Taste zu vermeiden. Es ist in Physical-Modeling-Strukturen aber oft störend, weil die Töne nach dem Loslassen der Taste nicht natürlich abklingen. Mit dem Envelope C mit langer Release-Zeit kann man das zwar umgehen, damit ist aber der Envelope für nicht anderes mehr verfügbar. Ausserdem ist er dafür "überqualifiziert".
Der neue negative Range eignet sich auch gut für die Anwendung eines Sustain/Damper-Pedals.

Der Parameter könnte vielleicht umbenannt werden in "FB - Gate/Env C".

### PR DESCRIPTION
**Pull request by schlenkibus**
closes #2921
closes #2920
closes #2919 
closes #2908 
closes #1563

### CODE REVIEW

File: projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/UnmodulatebaleParameterLayouts.cpp [hunk 1]
```diff
@@ -50,6 +58,17 @@ void UnmodulateableParameterSelectLayout2::init()
         addControl(new SelectedParameterKnubbelSlider(Rect(BIG_SLIDER_X, 24, BIG_SLIDER_WIDTH, 6)));
         break;
     }
+
+    m_isScaleParameter = ScaleGroup::isScaleParameter(p);
+    if(m_isScaleParameter)
+    {
+      {
+        auto bA = findControlOfType<SwitchVoiceGroupButton>();
+        remove(bA.get());
```
Comment (schlenkibus) [inline projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/UnmodulatebaleParameterLayouts.cpp:67]: check if bA is != nullptr

File: projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/UnmodulatebaleParameterLayouts.cpp [hunk 2]
```diff
@@ -59,6 +78,87 @@ void UnmodulateableParameterSelectLayout2::init()
   highlight<ParameterNameLabel>();
 }
 
+namespace detail
```
Comment (schlenkibus) [inline projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/UnmodulatebaleParameterLayouts.cpp:81]: move into Control itself, remove duplicated helpers

File: projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/UnmodulatebaleParameterLayouts.h [hunk 3]
```diff
@@ -21,7 +21,19 @@ class UnmodulateableParameterSelectLayout2 : public ParameterSelectLayout2, publ
   UnmodulateableParameterSelectLayout2();
 
  protected:
+  void onParameterSelectionChanged(Parameter* oldP, Parameter* newP);
```
Comment (schlenkibus) [inline projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/UnmodulatebaleParameterLayouts.h:24]: use private if applicable, and initialize new members

File: projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/controls/ParameterCarousel.cpp [hunk 4]
```diff
@@ -165,12 +187,71 @@ void ParameterCarousel::setupChildControlsForParameterWithoutButtonMapping(Param
...
+                                            C15::PID::Scale_Offset_7, C15::PID::Scale_Offset_8, C15::PID::Scale_Offset_9, C15::PID::Scale_Offset_10,
+                                            C15::PID::Scale_Offset_11, C15::PID::Master_Volume});
+  }
+  else if(paramID == C15::PID::Unison_Voices || paramID == C15::PID::Unison_Detune || paramID == C15::PID::Unison_Phase || paramID == C15::PID::Unison_Pan)
+  {
+    setupChildControls(selectedParameter, { C15::PID::Unison_Voices, C15::PID::Unison_Detune, C15::PID::Unison_Phase, C15::PID::Unison_Pan });
+  }
+}
+
+void ParameterCarousel::setupChildControlsForScaleParameterCarousel(Parameter* selectedParameter, const std::list<int>& buttonAssignments)
```
Comment (schlenkibus) [inline projects/epc/playground/src/proxies/hwui/panel-unit/boled/parameter-screens/controls/ParameterCarousel.cpp:207]: do technical refactoring, extracting functionality into named methods

File: projects/web/static/nonmaps/src/main/java/com/nonlinearlabs/client/world/maps/parameters/ModulateableSliderHorizontal.java [hunk 5]
```diff
@@ -41,10 +41,9 @@ private void drawModIndication(Context2d ctx) {
 
 		if (isBiPolar) {
 			double x = r.getCenterPoint().getX();
```
Comment (schlenkibus) [inline projects/web/static/nonmaps/src/main/java/com/nonlinearlabs/client/world/maps/parameters/ModulateableSliderHorizontal.java:43]: check if removed factor of 2 is correct

Comment (schlenkibus) [reply projects/web/static/nonmaps/src/main/java/com/nonlinearlabs/client/world/maps/parameters/ModulateableSliderHorizontal.java:43]: This is correct. If I revert this change the resulting modulation indication falls short of the expected width by an factor of 2. I believe this bug must have been here before and now it is fixed.

**Review by hhoegelo — APPROVED** [review]

**Pull request merged**